fdbclient/ReadYourWrites.h

```diff
@@ -54,6 +54,26 @@ struct TransactionDebugInfo : public ReferenceCounted<TransactionDebugInfo> {
 	TransactionDebugInfo() : transactionName(""), lastRetryLogTime() { }
 };
 
+// Private keyspace framework
+class PrivateKeyRangeImpl {
+public:
+	virtual Future<Standalone<RangeResultRef>> getRange(const KeyRange& keys, ReadYourWritesTransaction* ryw) = 0;
```

Comment:
I think this should be `const`.

Not sure whether Andrew already discussed this with you, but what do you think of having the original function here:
```
virtual Future< Standalone<RangeResultRef> > getRange( KeySelector begin, KeySelector end, GetRangeLimits limits, bool snapshot = false, bool reverse = false ) const = 0;
```
You than could create a subclass that implements this but also adds a virtual:
```
virtual Future<Standalone<RangeResultRef>> getRange(const KeyRange& keys, ReadYourWritesTransaction* ryw) const = 0;
```
so your implementation could use this. From there you can even add another subclass that implements this simpler one but adds a new method that just returns all kv-pairs - so something like:

```
virtual Standalone<RangeResultRef> get() const = 0;
```

Reply:
I think this works. 
The point I am not sure is that do we need the original interface in some cases
``` getRange(KeySelector begin, KeySelector end, GetRangeLimits limits, bool snapshot = false, bool reverse = false) ```
Since this `PrivateKeyRangeImpl` knows nothing about other key Ranges, if the `KeySelectorRef` is pointing outside the underlying key Range of this class, it is not well defined. 
Since we can handle `KeySelectorRef`, `GetRangeLimits` and all parameters in `PrivateKeySpace::getRange`, this `getRange` can be simplified to only need a `KeyRangeRef`.

I do not come up with some cases where we need to handle these original parameters in `PrivateKeyRangeImpl`. Thus I may prefer the simplified version `getRange(KeyRangeRef keys, ReadYourWritesTransaction* ryw)`.